src/react/workflow/TagsFilter.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  AddButton,
+  Buttons,
+  Header,
+  HeaderKeyTag,
+  HeaderValueTag,
+  InputTag,
+  Inputs,
+  Item,
+  Items,
+  SubButton,
+} from '../ui-elements/EditableKeyValue';
+import type { Tags, Tag } from '../../types/s3';
+import FormContainer, * as F from '../ui-elements/FormLayout';
+import { useFieldArray, useForm } from 'react-hook-form';
+import { useEffect } from 'react';
+const EMPTY_ITEM = {
+  key: '',
+  value: '',
+};
+
+const removeEmptyKeys = (tags: Tag[]) =>
+  tags
+    .filter((tag) => tag.key !== '');
+
+type Props = {
+  objectMetadata: { tags: Tags };
+  handleChange:  (data: Array<{key, value}>) => void;
+};
+
+type FormValues = {
+  tags: Tag[];
+};
+
+function TagsFilter({ objectMetadata, handleChange }: Props) {
```

Comment:
I would advise to not set again a use form in this component but more to rely on the already set parent form. The you can use https://react-hook-form.com/api/useformcontext#main